test/index-test.js

```diff
@@ -5,14 +5,19 @@ const got = require('got');
 describe('Server', () => {
     const myServer = new app.MyServer();
 
-    before(() => {
+    beforeEach(() => {
```

Comment:
`mocha` can handle Promises returned from both test *and hook* functions. So we can deal with `listen` & `close` asynchronicity directly in the hooks, without adding an additional tests.

Reply:
Good to know, thanks. I'll move these tests to before() and after() then.